reps/2023-10-13-accelerator-support.md

```diff
@@ -0,0 +1,399 @@
+# Ray Accelerator Support
+
+## Summary
+
+Holistic design of supporting accelerators in Ray.
+This also serves as the reference doc for contributors
+to add new accelerator support in the future.
+
+### General Motivation
+
+Nowadays, more and more specialized accelerators (e.g. TPU, HPU) are coming out to speed up AI workloads.
+Supporting those accelerators natively (similar to how Ray supports Nvidia GPUs) in Ray will increase Ray's adoption as the computing framework for scaling AI applications.
+
+### Should this change be within `ray` or outside?
+
+Inside `ray` project since we want to support those accelerators out of the box.
+
+## Stewardship
+
+### Required Reviewers
+The proposal will be open to the public, but please suggest a few experienced Ray contributors in this technical domain whose comments will help this proposal. Ideally, the list should include Ray committers.
+
+@pcmoritz, @ericl, @scv119
+
+### Shepherd of the Proposal (should be a senior committer)
+To make the review process more productive, the owner of each proposal should identify a **shepherd** (should be a senior Ray committer). The shepherd is responsible for working with the owner and making sure the proposal is in good shape (with necessary information) before marking it as ready for broader review.
+
+@scv119
+
+## Design
+
+### Ray Core
+
+#### API
+Users will use accelerators through Ray resources. For GPU accelerators (e.g. Nvidia, AMD, Intel GPUs), they will use the `GPU` resource name. For other accelerators, they will each have their own resource name (e.g. TPU accelerator has resource name `TPU`).
+
+| Resource    | Accelerator               |
+| ----------- | ------------------------- |
+| GPU         | Nvidia, Intel, AMD, Apple |
+| TPU         | TPU                       |
+| HPU         | Habana Gaudi              |
+| neuron_cores| AWS Trainium, Inferentia  |
```

Comment:
Can someone ELI5 what the difference between TPU, GPU and neuron_cores is? And what other _types_ of accelerators might be part of the triple dots?

My angle is, if these all do matrix multiplications, from a user perspective, why would I need to change my codebase to use another of these resources?

Reply:
Different accelerators have different programming models. E.g. Nvidia GPU uses cuda and TPU uses XLA so normally you need to change your codebase when you switch from one type of accelerator to another.